docs/source/github_workflow.rst

```diff
@@ -1,37 +1,91 @@
-Release Workflow
-==================
+Git and Release Workflow
+=========================
 
-When preparing a new release, developers need to be aware of the existence of multiple processes.
+When preparing a new release, developers must take care to correctly follow the release process.
 
-Branches
+Git Branches
 --------------
 
-* develop
-* master
-* release
+* **master**: This is the stable branch. Release tags are always on this branch and the latest release is always the head of this branch.
+* **develop**: This is the main development branch and can be ahead of the master branch.
+  One should never directly merge and push to develop but perform a pull request on GitHub.
+  The PR can only be merged if approved by another developer and all automatic tests pass.
+* **wip_xx**: Feature branches for feature number xx start with wip_xx (one can add a short name to the branch name).
+  They always branch off develop, and as explained above, are merged to develop via GitHub pull requests.
+* **release/vx.y.z**: These are release branches (for version vx.y.z). They always branch off from develop.
+  Once the release process is complete (see below), the release branch is merged to both master and develop via GitHub pull requests.
```

Comment:
I think this part can be added to Contributing.md

Reply:
even the whole file is interesting to put in.. could be top level contributing.md. not sure what is best, any idea ?

Reply:
Moved to top-level contributing.md